lib/index.js

```diff
@@ -83,10 +83,14 @@ PostHTML.prototype.process = function (tree, options) {
    */
   options = options || {}
 
-  if (options.parser) parser = options.parser
-  if (options.render) render = options.render
+  if (options.parser && typeof options.parser === 'function') {
+    parser = options.parser
+  }
+  if (options.render && typeof options.render === 'function') {
+    render = options.render
+  }
 
-  tree = options.skipParse ? tree : parser(tree)
+  tree = typeof tree === 'object' ? tree : parser(tree)
```

Comment:
So I don't think this is the right way to handle this situation. Without the type checks, if someone passed in, let's say, a number as the parser, it would throw an error right away saying the number is not a function - clear, explicit, and not confusing. With these changes, it would silently ignore the input without generating any error or warning that it was incorrect.

If you want to do runtime type checks, I think thats fine, but I would make it more thorough, more consistent, and use something like [joi](https://github.com/hapijs/joi) to provide good error messages if one of the type checks fail.

Also, removing `skipParse` is a breaking change, so make sure this goes out in a major version bump!